### DISCUSSION TITLE
Option to filter message history when pressing up/down arrow keys based on the content of the chat text box

### DISCUSSION THREAD

**Discussion by KleberPF** [Chatterino/chatterino2#5456] [Ideas]
If I have "Kappa" in the chat text box, arrow up/down should only show previous messages that also start with "Kappa"

**Comment by pajlada** [comment]
I would prefer to not add some setting for this, and would rather this be made available through a secondary hotkey or something like that, that would open maybe a `:-completion` list with your message history that you can choose from (or maybe more similar to CTRL+K?)

**Reply by KleberPF** [reply]
I could remap arrow up/down to this new hotkey, right? If so, I don't mind it.

**Reply by pajlada** [reply]
I guess that's something we could do too - just add an additional parameter to the "next message" and "previous message" hotkeys to allow for "use current message as context" or something

**Comment by Mm2PL** [comment]
I'd love for this to be a plugin but realistically that's not going to be easy to make as a plugin compared to just implementing it ourselves in c++.
